pydantic/_internal/_decorators.py

```diff
@@ -298,6 +385,10 @@ def build(model_dc: type[Any]) -> DecoratorInfos:  # noqa: C901 (ignore complexi
                         model_dc, cls_var_name=var_name, shim=None, info=info
                     )
                 setattr(model_dc, var_name, var_value.wrapped)
+        try:
+            setattr(model_dc, '__pydantic_decorators__', res)
+        except (TypeError, AttributeError):
```

Comment:
Can we check explicitly for cases where we cannot set `__pydantic_decorators__` here and avoid relying on exceptions? At least a comment explaining the motivation behind this would be helpful here, imo.

Reply:
Reworked a bit. There's a comment now and no try/except